apps/desktop/desktop_native/core/src/crypto/crypto.rs

```diff
@@ -37,3 +37,30 @@ pub fn encrypt_aes256(
 
     Ok(CipherString::AesCbc256_B64 { iv, data })
 }
+
+pub fn argon2(
+    secret: &[u8],
+    salt: &[u8],
+    iterations: u32,
+    memory: u32,
+    parallelism: u32,
+) -> Result<[u8; 32]> {
+    use argon2::*;
+
+    let params = Params::new(memory, iterations, parallelism, Some(32)).map_err(|e| {
+        KdfParamError::InvalidParams(format!("Argon2 parameters are invalid: {e}",))
+    })?;
+    let argon = Argon2::new(Algorithm::Argon2id, Version::V0x13, params);
+
+    let mut hash = [0u8; 32];
+    let _ = argon.hash_password_into(secret, &salt, &mut hash);
```

Comment:
```suggestion
argon.hash_password_into(secret, &salt, &mut hash)?;
```

With `let _ =` won't the error case of the `Result<()>` be suppressed here? 

I would recommend using the `?` operator or explicitly handling the error result.

Reply:
Passing through with ? + map err now